app/main.py

```diff
@@ -2,0 +1,20 @@
+from typing import Union
+from app.cafe import Cafe
+from app.errors import (
+    NotWearingMaskError,
+    VaccineError
+)
+
+
+def go_to_cafe(friends: list[dict[str, Union[str, bool]]], cafe: Cafe) -> str:
+    need_mask = 0
+    for friend in friends:
+        try:
+            cafe.visit_cafe(friend)
+        except NotWearingMaskError:
+            need_mask += 1
+        except VaccineError:
+            return "All friends should be vaccinated"
+    if need_mask:
```

Comment:
Code Efficiency: There is no need to compare `need_mask` to `0`. You can simply write `if need_mask:`. This will check if `need_mask` is not zero.